lib/Minion/Backend/MongoDB.pm

```diff
@@ -43,72 +70,260 @@ sub enqueue {
   my $doc = {
     args    => $args,
     created => DateTime->from_epoch(epoch => time),
-    delayed => DateTime->from_epoch(epoch => $options->{delay} ? time + $options->{delay} : 1),
+    delayed  => DateTime->now()->add(seconds => $options->{delay} // 0),
     priority => $options->{priority} // 0,
     state    => 'inactive',
-    task     => $task
+    task     => $task,
+    retries  => 0,
+    attempts => $options->{attempts} // 1,
+    notes    => $options->{notes} || {},
+    parents  => $options->{parents} || [],
+    queue    => $options->{queue} // 'default',
   };
 
-  my $oid = $self->jobs->insert($doc);
-  $self->notifications->insert({c => 'created'});
+  my $res = $self->jobs->insert_one($doc);
+  my $oid = $res->inserted_id;
+  $self->notifications->insert_one({c => 'created'});
   return $oid;
 }
 
 sub fail_job { shift->_update(1, @_) }
 
 sub finish_job { shift->_update(0, @_) }
 
-sub job_info { $_[0]->_job_info($_[0]->jobs->find_one({_id => MongoDB::OID->new(value => "$_[1]")})); }
+sub history {
+  my $self = shift;
+
+  my $dt_stop     = DateTime->now;
+  my $dt_start    = $dt_stop->clone->add(days => -1);
+  my $dt_span     = DateTime::Span->from_datetimes(
+    start => $dt_start,
+    end => $dt_stop
+  );
+  my $dt_set      = DateTime::Set->from_recurrence(
+    recurrence => sub { return $_[0]->truncate(to=>'hour')->add(hours => 1) }
+  );
+  my @dt_set      = $dt_set->as_list(span => $dt_span);
+  my %acc = (map {&_dtkey($_) => {
+      epoch    => $_->epoch(),
+      failed_jobs   => 0,
+      finished_jobs => 0
+  }} @dt_set);
+
+  my $match  = {'$match' => { finished => {'$gt' => $dt_start} } };
+  my $group  = {'$group' => {
+    '_id'  => {
+        hour    => {'$hour' => {date => '$finished'}},
+        day     => {'$dayOfYear' => {date => '$finished'}},
+        year    => {'$year' => {date => '$finished'}}
+    },
+    finished_jobs => {'$sum' => {
+        '$cond' => {if => {'$eq' => ['$state', 'finished']}, then => 1, else => 0}
+    }},
+    failed_jobs => {'$sum' => {
+        '$cond' => {if => {'$eq' => ['$state', 'failed']}, then => 1, else => 0}
+    }}
+  }};
+
+  my $cursor = $self->jobs->aggregate([$match, $group]);
+
+  while (my $doc = $cursor->next) {
+    my $dt_finished = new DateTime(
```

Comment:
`DateTime->new` instead of `new DateTime`

Reply:
Solved but I'd like new Object syntax :-)